### PR TITLE
Develop Users CRUD API Endpoint & Add Excel export to Survey results Bug solution.

### PR DESCRIPTION
**Pull request by mujibhalimi**
I am using API for the user CRUD operation, to test the application please replace and add these files.

### CODE REVIEW

**Comment by mujibhalimi** [comment]
@jamshidhashimi Saib, please replace and add these files to the project it's will work for the user's CRUD operations using API.

File: routes/web.php [hunk 1]
```diff
@@ -1,207 +1,210 @@
...
-        return redirect('page/21');
-    });
-    Route::get('/logout', function() {
-        Auth::logout();
-        return redirect('/home');
-    });
-    Route::get('/home', 'HomeController@index')->name('home');
-});
-/** OTHER PAGES THAT SHOULD NOT BE LOCALIZED **/
-Route::post('resources/favorite', 'ResourceController@resourceFavorite');
```
Comment (jamshidhashimi) [inline routes/web.php]: Why you removed these @mujibhalimi? When working on the API side, why bother with `web.php`?

**Comment by jamshidhashimi** [comment]
@mujibhalimi I appreciate your efforts, but making the CRUD api involves dealing with `api.php` and api resources. Why there are a lot of changes in the Blade views?

File: routes/web.php (hunk 1 above)
Comment (mujibhalimi) [reply routes/web.php]: I messaged there in FB, I explained the reason that I merged the files but it's rebased and rewritten all repo. then I made again the web.php file there is only one route defined for the user to show the create blade page. and I did not remove and interact with other routes. my git cmd app giving me the error, therefore I used the upload button. then occurred this.

**Comment by mujibhalimi** [comment]
> @mujibhalimi I appreciate your efforts, but making the CRUD API involves dealing with `api.php` and API resources. Why there are a lot of changes in the Blade views?

thanks,  I am using for user CRUD Operations the api.php and resources for the controller,maybe my method is change from the other. please see these api.php routes.
 `// #(http://localhost:8000/api/show_users)
Route::get('/show_users','UserController@listUsers'); 
// this api route storing the users information 
//#(http://localhost:8000/api/store_user)
Route::post('/store_user','UserController@insert_user');
// this api route deleting the selected user 
//#(http://localhost:8000/api/delete_user)
Route::get('/delete_user/{id}','UserController@user_delete');
// this api route reteriving the single user 
//#(http://localhost:8000/api/get_user/user_id)
Route::get('/get_user/{id}','UserController@user_get');
// this api route update the selected user information 
//#(http://localhost:8000/api/update_user/user_id)
Route::post('/update_user/{id}','UserController@user_update');`
here is the UserController.php file code that i change and write my own code for CRUD operations.
` public function create()
   //this function shows the create page for user, which was not in project, I created new for user creation.
    {
        
        DDLClearSession();
        $myResources = new Resource();
        $countries = $myResources->resourceAttributesList('taxonomy_term_data', 15);
        $provinces = $myResources->resourceAttributesList('taxonomy_term_data', 12)->all();
        return view('users.user_create', compact('countries', 'provinces'));
    }
 /* show users list */
    public function listUsers()
    {
        $show_user = User::join('user_roles', 'users.id', 'user_roles.user_id')->where('user_roles.user_id', '!=', 'users.id')->get();
        return UserResource::collection($show_user);
    }
    // inserting user to db
    public function insert_user(Request $request)
    {
        $save_user = new User();
        $save_user->username = $request['UserName'];
        $save_user->email = $request['Email'];
        $save_user->password = bcrypt($request['Password']);
        $save_user->status = $request['Status'];
        $save_user->save();
        //  current saved user id 
        $c_id = DB::table('users')->orderBy('id', 'desc')->first();
        DB::table('user_roles')->insert([
            'user_id' => $c_id->id,
            'role_id' => $request['Role']
        ]);
        DB::table('user_profiles')->insert([
            'user_id' => $c_id->id,
            'first_name' => $request['first_name'],
            'last_name' => $request['last_name'],
            'phone' => $request['phone'],
            'gender' => $request['gender'],
            'country' => $request['country'],
            'city' => $request['city']
        ]);
        return response()->json($save_user);
    }
    // this function delete the user from list
    public function user_delete($id)
    {
        $find_user = User::find($id);
        // find the photo and remove from directory
        $oldPhoto = $find_user->photo;
        if ($oldPhoto !== $find_user->photo) {
            $this->removeUserphoto($oldPhoto);
        }
        $find_user->delete();
        return response()->json($find_user);
    }
    // this function reterive single user
    public function user_get($id)
    {
        $find_user_edit = User::find($id);
        return response()->json($find_user_edit);
    }
    // this function update the user which reterived
    public function user_update(Request $request, $id)
    {
        $find_user_update = User::find($id);
        if ($request->filled('city')) {
            $city = $request->input('city');
        } elseif ($request->filled('city_other')) {
            $city = $request->input('city_other');
        } else {
            $city = NULL;
        }
        $find_user_update->username = $request['username'];
        $find_user_update->email = $request['email'];
        if ($request->filled('password')) {
            $find_user_update->password = Hash::make($request->input('password'));
        }
        $find_user_update->password = bcrypt($request['password']);
        $find_user_update->status = $request['status'];
        $userProfile = UserProfile::where('user_id', $id)->first();
        $userProfile->first_name = $request->input('first_name');
        $userProfile->last_name = $request->input('last_name');
        $userProfile->gender = $request->input('gender');
        $userProfile->country = $request->input('country');
        $userProfile->city = $city;
        $userProfile->phone = $request->input('phone');
        $userProfile->save();
        $userRole = UserRole::where('user_id', $id);
        $userRole->delete();
        $userRole = new UserRole();
        $userRole->user_id = $id;
        $userRole->role_id = $request->input('role');
        $userRole->save();
        $find_user_update->save();
        return response()->json($find_user_update);
    }
} `
the reasons that i brought changes in blade view,i used ajax request for api use there for it shows changes in blade,i can tell you that there is not many changes please use and replace these files to demo than you will find the change.
thanks

**Comment by jamshidhashimi** [comment]
@mujibhalimi I can't merge this PR. It is touching to many points in other areas of the application that will create problems and errors in our online website as soon as I merge it. The issue is very specific to being only creating API endpoints, that's it.

**Pull request closed without merging**